src/encoded/types/cart.py

```diff
@@ -87,12 +90,46 @@ def get_or_create_cart_by_user(context, request):
     user = request.registry[COLLECTIONS]['user'].get(userid)
     if not user:
         raise HTTPBadRequest()
-    carts = _get_carts_by_user(request, userid)
-    if not carts:
-        cart = _create_cart(request, user)
+    cart_name = None
+    all_carts = get_cart_objects_by_user(request, userid)
+    countable_carts = [cart for cart in all_carts if cart['status'] != 'deleted' and cart['status'] != 'disabled']
```

Comment:
Sometimes easier to read if you split into multiple lines:
```python
[
    cart
    for cart in all_carts
    if cart['status'] not in ['deleted',  'disabled']
]
```

Reply:
Changed.

Reply:
thx, I'm pushing this format too.  They tend to get long and it's easier to add more conditions later on.